src/pudl/transform/phmsagas.py

```diff
@@ -0,0 +1,670 @@
+        MELT_PATTERNS["_core_phmsagas__yearly_distribution_leaks"]["services_pattern"],
+        {
+            "leak_severity": r"^(all_leaks|hazardous_leaks)",
+            "leak_source": r"^(?:all_leaks|hazardous_leaks)_(.*)_(?:mains|services)",
+        },
+        # All of the other tables with mains in it have the miles of main except for
+        # this one. We could totally edit _melt_merge_main_services to enable different
+        # column names instead of doing this rename
+    ).rename(columns={"mains_miles": "mains"})
+
+
+@asset(io_manager_key="pudl_io_manager", compute_kind="pandas")
+def _core_phmsagas__yearly_distribution_excavation_damages(
+    _core_phmsagas__yearly_distribution: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform table of damages - broken out by type and sub-type."""
+    damage_pattern = MELT_PATTERNS[
+        "_core_phmsagas__yearly_distribution_excavation_damages"
+    ]["damage_pattern"]
+    col_patterns = {
+        "damage_type": rf"excavation_damage_({'|'.join(DAMAGE_TYPES_PHMSAGAS)})",
+        "damage_sub_type": damage_pattern,
+    }
+    return _core_phmsagas__yearly_distribution.pipe(
+        _melt_col_pattern,
+        damage_pattern,
+        value_name="damages",
+        col_patterns=col_patterns,
+    ).reset_index()
+
+
+@asset(io_manager_key="pudl_io_manager", compute_kind="pandas")
+def _core_phmsagas__yearly_distribution_misc(
+    _core_phmsagas__yearly_distribution: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform this distribution table of miscellaneous numeric values."""
+    idx = ["report_date", "report_id", "operator_id_phmsa", "operating_state"]
+    df = (
+        _core_phmsagas__yearly_distribution.loc[
+            :, idx + YEARLY_DISTRIBUTION_MISC_COLUMNS
+        ]
+        .sort_values(YEARLY_DISTRIBUTION_MISC_COLUMNS, ascending=False)
+        # Convert percent unaccounted for gas to a fraction
+        .assign(
+            unaccounted_for_gas_fraction=lambda x: x.unaccounted_for_gas_percent / 100
+        )
+        .drop(columns=["unaccounted_for_gas_percent"])
+        .convert_dtypes()
+    )
+    return df
+
+
+@asset_check(asset=_core_phmsagas__yearly_distribution_misc, blocking=True)
+def _check_unaccounted_for_gas_fraction(df):
+    """Check what percentage of unaccounted gas values are reported as a negative number.
+
+    This is technically impossible but allowed by PHMSA.
+    """
+    # Count the rows where unaccounted gas is negative.
+    negative_count = (df["unaccounted_for_gas_fraction"] < 0).sum()
+
+    # Calculate the percentage
+    negative_percentage = negative_count / len(df)
+    if negative_percentage > 0.17:
+        error = f"Percentage of rows with negative unaccounted_for_gas_fraction values: {negative_percentage:.2f}"
+        logger.info(error)
+        return AssetCheckResult(passed=False, metadata={"errors": error})
+
+    return AssetCheckResult(passed=True)
```

Comment:
i had to up the % here to 17% for the fast tests...


with @krivard 's help I tried to add this as a dbt_expectations... we were hopeful bc [it looked like](https://greatexpectations.io/expectations/expect_column_values_to_be_between/) we could use this `mostly` arg.... but the [dbt version](https://www.elementary-data.com/dbt-tests/expect-column-values-to-be-between) doesn't have this `mostly` argument... sad. 

```python
          - name: unaccounted_for_gas_fraction
            data_tests:
              - dbt_expectations.expect_column_values_to_be_between:
                  arguments:
                    min_value: 0.0
                    mostly: .85
```